rpc/backend/tx_info.go

```diff
@@ -310,7 +324,7 @@ func (b *Backend) GetTransactionReceipt(hash common.Hash) (map[string]interface{
 
 			// Inclusion information: These fields provide information about the inclusion of the
 			// transaction corresponding to this reciept
-			"blockHash":        txDetails.TxReceipt.TxHash,
+			"blockHash":        common.BytesToHash(block.Block.Header.Hash()).Hex(),
```

Comment:
No need to fetch the block again just use `txDetails.TxReceipt.BlockHash`, I think this was a typo on my end.